src/mpid/ch4/netmod/ucx/ucx_win.h

```diff
@@ -41,15 +43,21 @@ static inline int MPIDI_UCX_Win_allgather(MPIR_Win * win, size_t length,
     ucp_context_h ucp_context = MPIDI_UCX_global.context;
 
     MPIDI_UCX_WIN(win).info_table = MPL_malloc(sizeof(MPIDI_UCX_win_info_t) * comm_ptr->local_size);
+    mem_map_params.field_mask = UCP_MEM_MAP_PARAM_FIELD_ADDRESS |
+                                UCP_MEM_MAP_PARAM_FIELD_LENGTH|
+                                UCP_MEM_MAP_PARAM_FIELD_FLAGS;
     if (length == 0)
-        base = &ucx_dummy_buffer;
+        mem_map_params.address = &ucx_dummy_buffer;
     else
-        base = *base_ptr;
+        mem_map_params.address = *base_ptr;
+     mem_map_params.length = size;
+     mem_map_params.flags =  UCP_MEM_MAP_NONBLOCK ;
 
```

Comment:
Set it to nonblocking.
This may make the first put/get a little bit slower - but could save some system resources.